pkg/pillar/agentlog/agentlog.go

```diff
@@ -135,7 +143,8 @@ func handleSignals(sigs chan os.Signal) {
 // Print out our stack
 func printStack() {
 	stacks := getStacks(false)
-	log.Errorf("fatal stack trace due to %s:\n%v\n", savedRebootReason, stacks)
+	logStacks(fmt.Sprintf("fatal stack trace due to %s:\n\n %v\n",
```

Comment:
Why do you need to split this into two? And why change from log.Error to log.Warn? Can keep this as it is AFAICT